### PR TITLE
Update envtest to support build on M1 machine

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
- I needed to update the ENVTEST_K8S_VERSION flag to make it possible to run `make` on m1 mac.
- I also need to replace `go get` with `go install` to ensure `Makefile` working. (local `go version go1.18.3 darwin/arm64`)

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
hmmm looks like that version doesn't quite work. @brucez-anyscale how did you get the `make` running on m1?

**Comment by brucez-anyscale** [comment]
> hmmm looks like that version doesn't quite work. @brucez-anyscale how did you get the `make` running on m1?

I think the go version is still 1.17 for kuberay. I do not know special things for local m1 build.
Only the unit test will not work on m1, but build should work.

File: ray-operator/Makefile [hunk 1]
```diff
@@ -115,8 +115,8 @@ set -e ;\
 TMP_DIR=$$(mktemp -d) ;\
 cd $$TMP_DIR ;\
 go mod init tmp ;\
-echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+echo "Downloading $(2) to $(PROJECT_DIR)/bin" ;\
+GOBIN=$(PROJECT_DIR)/bin go install $(2) ;\
```
Comment (brucez-anyscale) [inline ray-operator/Makefile:119]: this is go 1.17 and go 1.18 difference.

**Comment by simon-mo** [comment]
ahhh i see the issue now; i'll downgrade to go 1.17

**Pull request closed without merging**